File: packages/elixir-sdk/lib/tree_db_sdk/adapters/graph.ex
defmodule TreeDbSdk.Graph do
  @moduledoc false
  alias TreeDbSdk.Adapters.Common

  def refresh(client, repo_id, body \\ %{}) do
    Common.json_request(
      client,
      :post,
      "/api/v1/repos/" <> Common.segment(repo_id) <> "/graph/refresh",
      body,
      %{}
    )
  end

  def query(client, repo_id, body) do
    Common.json_request(
      client,
      :post,
      "/api/v1/repos/" <> Common.segment(repo_id) <> "/graph/query",
      body,
      %{}
    )
  end
end

File: packages/python-sdk/src/treedb_sdk/adapters/graph.py
from __future__ import annotations

from typing import Any

from .common import json_request, segment
from treedb_sdk.transport import Transport


class GraphAdapter:
    def __init__(self, transport: Transport) -> None:
        self.transport = transport

    def refresh(self, repo_id: str, body: Any | None = None) -> Any:
        return json_request(self.transport, "POST", f"/api/v1/repos/{segment(repo_id)}/graph/refresh", body)

    def query(self, repo_id: str, body: Any) -> Any:
        return json_request(self.transport, "POST", f"/api/v1/repos/{segment(repo_id)}/graph/query", body)
